src/EventStore.Core.Tests/Services/Transport/Grpc/StreamsTests/DeleteTests.cs

```diff
@@ -0,0 +1,151 @@
+						Count = 1,
+						Stream = new() {
+							StreamIdentifier = new() { StreamName = ByteString.CopyFromUtf8(StreamName) },
+							Start = new()
+						}
+					}
+				}, GetCallOptions());
+
+				var responses = await call.ResponseStream.ReadAllAsync().ToArrayAsync();
+
+				Assert.AreEqual(1, responses.Length);
+				Assert.AreEqual(new ReadResp {
+					StreamNotFound = new() {
+						StreamIdentifier = new() { StreamName = ByteString.CopyFromUtf8(StreamName) }
+					}
+				}, responses[0]);
+			}
+
+			protected override async Task Given() => _appendResponse = await AppendToStreamBatch(new BatchAppendReq {
+				Options = new() {
+					NoStream = new(),
+					StreamIdentifier = new() { StreamName = ByteString.CopyFromUtf8(StreamName) }
+				},
+				CorrelationId = Uuid.NewUuid().ToDto(),
+				IsFinal = true,
+				ProposedMessages = { CreateEvents(1) }
+			});
+
+			protected override async Task When() {
+				using var call = StreamsClient.DeleteAsync(new() {
+					Options = _options
+				}, GetCallOptions());
+
+				try {
+					_deleteResponse = await call.ResponseAsync;
+				} catch (Exception ex) {
+					_caughtException = ex;
+				}
+			}
+		}
+
+		[TestFixture(typeof(LogFormat.V2), typeof(string))]
+		[TestFixture(typeof(LogFormat.V3), typeof(uint))]
+		public class DeleteExistingStreamExpectedRevision<TLogFormat, TStreamId>
+			: DeleteExistingStreamSpecification<TLogFormat, TStreamId> {
+			public DeleteExistingStreamExpectedRevision() : base(StreamRevision.Start) { }
+		}
+
+		[TestFixture(typeof(LogFormat.V2), typeof(string))]
+		[TestFixture(typeof(LogFormat.V3), typeof(uint))]
+		public class DeleteExistingStreamAny<TLogFormat, TStreamId>
+			: DeleteExistingStreamSpecification<TLogFormat, TStreamId> {
+			public DeleteExistingStreamAny() : base(ExpectedVersion.Any) { }
+		}
+
+		[TestFixture(typeof(LogFormat.V2), typeof(string))]
+		[TestFixture(typeof(LogFormat.V3), typeof(uint))]
+		public class DeleteExistingStreamExists<TLogFormat, TStreamId>
+			: DeleteExistingStreamSpecification<TLogFormat, TStreamId> {
+			public DeleteExistingStreamExists() : base(ExpectedVersion.StreamExists) { }
+		}
+
+		[TestFixture(typeof(LogFormat.V2), typeof(string))]
+		[TestFixture(typeof(LogFormat.V3), typeof(uint))]
+		public class DeleteNoStream<TLogFormat, TStreamId> : GrpcSpecification<TLogFormat, TStreamId> {
+			private Exception _caughtException;
+			protected override Task Given() => Task.CompletedTask;
+
+			protected override async Task When() {
+				using var call = StreamsClient.DeleteAsync(new() {
+					Options = new() {
+						NoStream = new(),
+						StreamIdentifier = new() {
+							StreamName = ByteString.CopyFromUtf8(Guid.NewGuid().ToString())
+						}
+					}
+				}, GetCallOptions());
+
+				try {
+					await call.ResponseAsync;
+				} catch (Exception ex) {
+					_caughtException = ex;
+				}
+			}
+
+			[Test]
+			public void an_exception_is_thrown() {
+				if (_caughtException is not RpcException ex) {
```

Comment:
I wonder why IsInstanceOf doesn't return the object cast to the specified type

anyway, style thing, but perhaps this would be easier to read:

```
			[Test]
			public void an_exception_is_thrown() {
				Assert.IsInstanceOf<RpcException>(_caughtException);
				var ex = _caughtException as RpcException;
				Assert.AreEqual(StatusCode.FailedPrecondition, ex.StatusCode);
			}
```